src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/CreationEffects.cs

```diff
@@ -1,24 +1,38 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+#nullable enable
+
 using System.Collections.Generic;
 using Microsoft.TemplateEngine.Abstractions;
 
 namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects
 {
     internal class CreationEffects : ICreationEffects
     {
-        public IReadOnlyList<IFileChange> FileChanges { get; set; }
+        internal CreationEffects(IReadOnlyList<IFileChange> fileChanges, ICreationResult creationResult)
+        {
+            FileChanges = fileChanges ?? throw new System.ArgumentNullException(nameof(fileChanges));
+            CreationResult = creationResult ?? throw new System.ArgumentNullException(nameof(creationResult));
+        }
+
+        public IReadOnlyList<IFileChange> FileChanges { get; }
 
-        public ICreationResult CreationResult { get; set; }
+        public ICreationResult CreationResult { get; }
     }
 
     internal class CreationEffects2 : ICreationEffects, ICreationEffects2
     {
-        public IReadOnlyList<IFileChange2> FileChanges { get; set; }
+        internal CreationEffects2(IReadOnlyList<IFileChange2> fileChanges, ICreationResult creationResult)
+        {
+            FileChanges = fileChanges ?? throw new System.ArgumentNullException(nameof(fileChanges));
```

Comment:
nit: I wonder if its worth adding this null checks for API that is internal, since in theory, "soon" anything calling this will be "nullability" aware...